pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -439,9 +469,50 @@ func (c *Controller) HasSynced() bool {
 		!c.nodeInformer.HasSynced() {
 		return false
 	}
+
+	// after informer caches sync the first time, process resources in order
+	c.syncMu.Lock()
+	defer c.syncMu.Unlock()
+	if !c.initialSync {
+		if err := c.SyncAll(); err != nil {
```

Comment:
Note: even if all informers have synced, we can not make sure all resources are populated.

Reply:
If the informers have synced, that means the index has been synced. We're just listing the index in SyncAll.

Reply:
Looks correct from `func (p *ListPager) List(ctx context.Context, options metav1.ListOptions) `, I misunderstood it.